### DISCUSSION TITLE
Where to find the exact fasta files used for model building

### DISCUSSION THREAD

**Discussion by hamidghaedi** [brianhie/viral-mutation#9] [Q&A]
Hi there,
I really enjoyed reading the paper and would love to read it again.
Reading the paper supplementary document on page 3, I can see "We trained an amino acid residue-level language model on a total of 44,851 unique influenza A hemagglutinin (HA) amino acid sequences observed in animal
hosts from 1908 through 2019." I thought this is the number of sequences present in the "data/influenza/ird_influenzaA_HA_allspecies.fa" file. But that was not true as ird_influenzaA_HA_allspecies.fa file contains 94,560 sequences. The same story is true for other viruses and number of records in provided fasta files are not the same to what you mentioned in the supplementary file.
Would you please let me know where to look to see the exact fasta files you have used to train the models?

**Comment by brianhie** [comment]
Hi @hamidghaedi, we filter out some sequences and also only keep unique sequences, see here: https://github.com/brianhie/viral-mutation/blob/6c463b1744038a6d3e5927fe8a3270f9cd51fcbc/bin/flu.py#L60. The fasta file in the data/ folder will have some duplicate entries, etc. Once the fasta file has been processed, there should be the same number of sequences mentioned in the supplement. If you want a fasta file containing the training set, you could call the `process()` function and write it to a file.
